source/common/router/router.cc

```diff
@@ -2132,10 +2132,19 @@ void Filter::maybeProcessOrcaLoadReport(const Envoy::Http::HeaderMap& headers_or
 
   orca_load_report_received_ = true;
 
-  ENVOY_STREAM_LOG(trace, "Adding ORCA load report {} to load metrics", *callbacks_,
-                   orca_load_report->DebugString());
-  Envoy::Orca::addOrcaLoadReportToLoadMetricStats(
-      cluster_->lrsReportMetricNames().value(), orca_load_report.value(), host->loadMetricStats());
+  if (need_to_send_load_report) {
+    ENVOY_STREAM_LOG(trace, "Adding ORCA load report {} to load metrics", *callbacks_,
+                     orca_load_report->DebugString());
+    Envoy::Orca::addOrcaLoadReportToLoadMetricStats(cluster_->lrsReportMetricNames().value(),
+                                                    orca_load_report.value(),
+                                                    host->loadMetricStats());
+  }
+  if (orca_load_report_callbacks_.has_value()) {
+    const absl::Status status = orca_load_report_callbacks_->onOrcaLoadReport(*orca_load_report);
+    if (!status.ok()) {
+      ENVOY_LOG_MISC(error, "Failed to invoke OrcaLoadReportCallbacks: {}", status.message());
```

Comment:
consider using `ENVOY_LOG/ENVOY_STREAM_LOG` rather than the misc one.

Reply:
@efimki  Merge it due to tight schedule but please do address this comment above in a follow-up.

Reply:
Thanks, @tyxia! My apologies as I've totally missed this comment. I'll address that in a follow-up.